lib/client/bundler/proto.js

```diff
@@ -66,145 +125,250 @@ module.exports = function(ss, bundlers, bundlerById, options) {
+   * Multiple arguments will attempt to add module for the name.
+   * Special module names are 'loader' and 'start'.
+   *
+   * @returns {AssetEntry} Resource entries array
+   */
+  function systemModule(/* first_name */) {
+    var entries = [],
+        // jshint validthis:true
+        client = this.client;
+
+    /*jshint -W084 */
+    for(var i= 0,name; name = arguments[i]; ++i) {
+      switch(name) {
+        case 'loader':
+          entries.push(ss.bundler.browserifyLoader());
+          break;
+        case 'libs':
+          entries.push(ss.bundler.systemLibs());
+          break;
+        case 'start':
+          entries = entries.concat(ss.bundler.constants(client));
+          if (client.includes.initCode) {
+            entries = entries.concat(ss.bundler.startCode(client));
+          }
+          break;
 
-      // mixin client constants
-      var client = this.client;
-      if (client.constants) {
-        for(k in client.constants) {
-          result[k] = client.constants[k];
-        }
+        //case "eventemitter2":
+        //case "socketstream":
+        default:
+          entries.push(ss.bundler.systemModule(name));
+          break;
       }
+    }
 
-      return result;
-    },
+    return entries;
+  }
 
-    locals: function() {
-      var result = {}, k;
+  function constants() {
+    var result = {}, k;
 
-      // mixin system locals
-      for(k in system.assets.locals) {
-        result[k] = system.assets.locals[k].value;
-      }
+    // mixin system constants
+    for(k in system.assets.constants) {
+      result[k] = system.assets.constants[k].value;
+    }
 
-      // mixin client constants
-      var client = this.client;
-      if (client.locals) {
-        for(k in client.locals) {
-          result[k] = client.locals[k];
-        }
+    // mixin client constants
+    // jshint validthis:true
+    var client = this.client;
+    if (client.constants) {
+      for(k in client.constants) {
+        result[k] = client.constants[k];
       }
+    }
 
-      return result;
-    },
+    return result;
+  }
 
-    htmlTags: function(type,pack) {
-      var dests;
+  function locals() {
+    var result = {}, k;
 
-      if (type === 'start') {
-        return ['<script>' + this.module('start').map(function(value) { return value.content; }).join('\n') + '</script>'];
-      }
+    // mixin system locals
+    for(k in system.assets.locals) {
+      result[k] = system.assets.locals[k].value;
+    }
 
-      if (pack) {
-        dests = { urls: {} };
-        dests.urls[type] = resolveAssetLink(this.client, type);
-        return [ htmlTag[type](dests) ];
-      } else {
-        var entries = this.entries(type);
-        dests = this.dests;
-        return entries.map(function(entry) {
-          return htmlTag[type](dests,entry);
-        });
+    // mixin client constants
+    // jshint validthis:true
+    var client = this.client;
+    if (client.locals) {
+      for(k in client.locals) {
```

Comment:
The body of a for in should be wrapped in an if statement to filter unwanted properties from the prototype.